### PR TITLE
Implement account per instance feature

### PR DESCRIPTION
**Pull request by urFate**
Implement account per instance feature (#1127)

<details>
  <summary><h2>Screenshots</h2></summary>

> <h3>Out of box</h3>
> 

> <h3>Microsoft account</h3>
> 

> <h3>Offline account</h3>
> 

</details>

### CODE REVIEW

**Comment by BowDown097** [comment]
Good idea. I have some recommendations before the code gets merged:
- The override account should be acquired by ID, not name, in the edge case an offline account and online account share the same name.
- I'm pretty sure the OverrideAccount setting is completely unnecessary. Couldn't we just check if the OverrideAccountName setting exists and/or is not empty?

**Comment by urFate** [comment]
> - I'm pretty sure the OverrideAccount setting is completely unnecessary. Couldn't we just check if the OverrideAccountName setting exists and/or is not empty?

Combo box is not always empty and can contain data which saves to config anyway. For example player choosed account, then saved it, and disabled checkbox. In this case we can not rely on OverrideAccountName.

File: launcher/ui/pages/instance/InstanceSettingsPage.ui [hunk 1]
```diff
@@ -39,7 +39,7 @@
       <enum>QTabWidget::Rounded</enum>
      </property>
      <property name="currentIndex">
-      <number>0</number>
+      <number>5</number>
```
Comment (Sneedplex) [inline launcher/ui/pages/instance/InstanceSettingsPage.ui]: This should be 0.

File: launcher/ui/pages/instance/InstanceSettingsPage.ui [hunk 2]
```diff
@@ -595,6 +595,48 @@
          </layout>
         </widget>
        </item>
+       <item>
+        <widget class="QGroupBox" name="accountGroupBox">
+         <property name="title">
+          <string>Set a account to use</string>
```
Comment (Sneedplex) [inline launcher/ui/pages/instance/InstanceSettingsPage.ui]: an*

**Review by Sneedplex — CHANGES_REQUESTED** [review]
A few small nitpicks but LGTM.

File: launcher/ui/pages/instance/InstanceSettingsPage.ui (hunk 1 above)
Comment (urFate) [reply launcher/ui/pages/instance/InstanceSettingsPage.ui]: Ok. I didn't saw this moment.

File: launcher/ui/pages/instance/InstanceSettingsPage.ui (hunk 2 above)
Comment (urFate) [reply launcher/ui/pages/instance/InstanceSettingsPage.ui]: English is not my native. I'm too lazy to check grammar.

**Comment by LennyMcLennington** [comment]
@urFate sorry for being silent about this, I'm gonna test and review it today.

**Comment by LennyMcLennington** [comment]
I've changed it to use the profile ID instead of the index, since the index is not reliable (like if you have accounts A, B, C, D, and you select account C, the index will be 2, if you then delete account B, then your account like will then be A, C, D and your instance would suddenly be launching with account D instead)

**Comment by LennyMcLennington** [comment]
@urFate if you're happy with the changes I made and if the checks all complete successfully then I'll merge the PR. I will wait to hear back from you.

**Comment by urFate** [comment]
> @urFate if you're happy with the changes I made and if the checks all complete successfully then I'll merge the PR. I will wait to hear back from you.

Okay 👌

**Pull request merged**